ddp-workspace/projects/ddp-dsm-ui/src/app/activity-data/services/buildActivityData.service.ts

```diff
@@ -14,10 +14,13 @@ export class buildActivityDataService {
     constructor(private buildingFactory: BuildingFactoryService) {}
 
     public buildActivity(activityDefinition: ActivityDefinition, activity: ActivityData): QuestionTypeModel[] {
-      for(const answer of activity.questionsAnswers) {
-        !([answer?.stableId, activityDefinition?.questions, activityDefinition.getQuestionDefinition(answer.stableId)]
-          .includes(null)) && this.checkQuestionType(answer, activityDefinition.getQuestionDefinition(answer.stableId));
-      }
+      activityDefinition?.questions.forEach(question => {
+        const questionAnswer = activity.checkAnswerStableId(question.stableId);
+
+        questionAnswer?.stableId && question ? this.checkQuestionType(questionAnswer, question)
```

Comment:
do we need `&& question`? I believe if question does not exist for each won't iterate.